TimeSeriesLifecycleActionsIT#testForceMergeActionWithCompressionCodec can fail with assertion error

**Build scan**: https://gradle-enterprise.elastic.co/s/5wksambdfququ

**Repro line**:
```
./gradlew ':x-pack:plugin:ilm:qa:multi-node:javaRestTest' --tests "org.elasticsearch.xpack.ilm.TimeSeriesLifecycleActionsIT.testForceMergeActionWithCompressionCodec" -Dtests.seed=769BF75C3D24820D -Dtests.security.manager=true -Dtests.locale=ar-JO -Dtests.timezone=America/Indiana/Marengo -Druntime.java=11
```

**Reproduces locally?**: no.

**Applicable branches**: master, 7.x

**Failure history**: seems to have started failing occasionally starting on November 3rd.

**Failure excerpt**:
```
org.elasticsearch.xpack.ilm.TimeSeriesLifecycleActionsIT > testForceMergeActionWithCompressionCodec FAILED |  
-- | --
  | java.lang.AssertionError: |  
  | Expected: <1> |  
  | but: was <3> |  
  | at __randomizedtesting.SeedInfo.seed([769BF75C3D24820D:5F6B333BE1107442]:0) |  
  | at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18) |  
  | at org.junit.Assert.assertThat(Assert.java:956) |  
  | at org.junit.Assert.assertThat(Assert.java:923) |  
  | at org.elasticsearch.xpack.ilm.TimeSeriesLifecycleActionsIT.lambda$forceMergeActionWithCodec$34(TimeSeriesLifecycleActionsIT.java:496) |  
  | at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:950) |  
  | at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:923) |  
  | at org.elasticsearch.xpack.ilm.TimeSeriesLifecycleActionsIT.forceMergeActionWithCodec(TimeSeriesLifecycleActionsIT.java:493) |  
  | at org.elasticsearch.xpack.ilm.TimeSeriesLifecycleActionsIT.testForceMergeActionWithCompressionCodec(TimeSeriesLifecycleActionsIT.java:507)
```

--- a/x-pack/plugin/ilm/qa/multi-node/src/javaRestTest/java/org/elasticsearch/xpack/ilm/TimeSeriesLifecycleActionsIT.java
+++ b/x-pack/plugin/ilm/qa/multi-node/src/javaRestTest/java/org/elasticsearch/xpack/ilm/TimeSeriesLifecycleActionsIT.java
@@ -25,6 +25,7 @@ import org.elasticsearch.common.xcontent.XContentHelper;
 import org.elasticsearch.common.xcontent.XContentType;
 import org.elasticsearch.common.xcontent.json.JsonXContent;
 import org.elasticsearch.index.IndexSettings;
+import org.elasticsearch.index.engine.EngineConfig;
 import org.elasticsearch.snapshots.SnapshotState;
 import org.elasticsearch.test.rest.ESRestTestCase;
 import org.elasticsearch.xpack.cluster.routing.allocation.DataTierAllocationDecider;
@@ -493,7 +494,7 @@ public class TimeSeriesLifecycleActionsIT extends ESRestTestCase {
         assertBusy(() -> {
             assertThat(getStepKeyForIndex(client(), index), equalTo(PhaseCompleteStep.finalStep("warm").getKey()));
             Map<String, Object> settings = getOnlyIndexSettings(client(), index);
-            assertThat(getNumberOfSegments(client(), index), equalTo(1));
+            assertThat(settings.get(EngineConfig.INDEX_CODEC_SETTING.getKey()), equalTo(codec));
             assertThat(settings.get(IndexMetadata.INDEX_BLOCKS_WRITE_SETTING.getKey()), equalTo("true"));
         });
         expectThrows(ResponseException.class, () -> indexDocument(client(), index));
@@ -503,7 +504,6 @@ public class TimeSeriesLifecycleActionsIT extends ESRestTestCase {
         forceMergeActionWithCodec(null);
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/65171")
     public void testForceMergeActionWithCompressionCodec() throws Exception {
         forceMergeActionWithCodec("best_compression");
     }

